mvpa2/clfs/gda.py

```diff
@@ -259,3 +259,94 @@ def _g_k(self, data):
             dm = data - m
             res.append(b - 0.5 * np.sum(np.dot(dm, covi) * dm, axis=1))
         return np.array(res).T
+
+class ShrinkageLDA(GDA):
+    """LDA with Ledoit-Wolf estimate of covariance matrix
+    """
+
+    __tags__ = GDA.__tags__ + ['linear', 'lda', 'shrinkage']
+
+    def __init__(self, shrinkage_estimator='LedoitWolf', block_size=1000, \
+    **kwargs):
+        super(ShrinkageLDA, self).__init__(**kwargs)
+        self.block_size = block_size
+        if shrinkage_estimator is 'LedoitWolf':
+            self.shrinkage_estimator = LedoitWolf(block_size=block_size)
+        elif shrinkage_estimator is 'OAS':
+            self.shrinkage_estimator = OAS()
+        else:
+            raise ValueError, "Unknown estimator '%s'" % shrinkage_estimator
+
+    def _train(self, dataset):
+        """Train the classifier using `dataset` (`Dataset`).
+        """
+
+        # Adapted from GDA's method _train. Couldn't use that implementation
+        # because it only computes the class-dependent SSCP matrices. The
+        # shrinkage coefficient, however, needs to be computed from the whole
+        # time series (with samples centered on their respective class means)
+        params = self.params
```

Comment:
here we go -- that is where you would find those `Parameter`s after you move them out to be defined at the class level, e.g. `params.shrinkage_estimator`